sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -568,7 +577,22 @@ private void consumerPollLoop() {
       while (!closed.get()) {
         try {
           if (records.isEmpty()) {
+            // Each source has a single unique topic.
+            List<TopicPartition> topicPartitions = source.getSpec().getTopicPartitions();
+            Preconditions.checkStateNotNull(topicPartitions);
+            String topicName = "null"; // value will be overridden
+            for (TopicPartition topicPartition : topicPartitions) {
+              topicName = topicPartition.topic();
+              break;
+            }
+
+            java.time.Instant operationStartTime = java.time.Instant.now();
```

Comment:
do we need the start/end or just elapsed? If so I think https://guava.dev/releases/19.0/api/docs/com/google/common/base/Stopwatch.html is preferred as it takes care that it is monotonic etc.

Reply:
Just need elapsed, done.